workspaces/services/guacamole-azure-dsvm/porter.yaml

```diff
@@ -0,0 +1,83 @@
+name: tre-service-guacamole-win-dsvm
+version: 0.1.2
+description: "An Azure TRE target resource service for Guacamole (Windows DSVM)"
+registry: azuretre
+dockerfile: Dockerfile.tmpl
+
+credentials:
+  - name: azure_tenant_id
+    env: ARM_TENANT_ID
+  - name: azure_subscription_id
+    env: ARM_SUBSCRIPTION_ID
+  - name: azure_client_id
+    env: ARM_CLIENT_ID
+  - name: azure_client_secret
+    env: ARM_CLIENT_SECRET
+
+parameters:
+  - name: workspace_id
+    type: string
+  - name: tre_id
+    type: string
+  - name: mgmt_resource_group_name
+    type: string
+    description: "Resource group containing the shared ACR"
+    env: MGMT_RESOURCE_GROUP_NAME
+  - name: tfstate_resource_group_name
+    type: string
+    description: "Resource group containing the Terraform state storage account"
+  - name: tfstate_storage_account_name
+    type: string
+    description: "The name of the Terraform state storage account"
+  - name: tfstate_container_name
+    env: tfstate_container_name
+    type: string
+    default: "tfstate"
+    description: "The name of the Terraform state storage container"
+
+outputs:
+  - name: guacamole_win_dsvm_name
+    type: string
+
+mixins:
+  - exec
+  - terraform:
+      clientVersion: 0.15.4
+
+install:
+  - terraform:
+      description: "Deploy Guacamole Target Resource Service (Windows DSVM)"
+      vars:
+        workspace_id: "{{ bundle.parameters.workspace_id }}"
+        tre_id: "{{ bundle.parameters.tre_id }}"
+        arm_client_id: "{{ bundle.credentials.azure_client_id }}"
+        arm_client_secret: "{{ bundle.credentials.azure_client_secret }}"
```

Comment:
We should be using MSI for deploying. If look at https://github.com/microsoft/AzureTRE/blob/develop/workspaces/vanilla/porter.yaml - have an ARM_USE_MSI variable. nothing is being passed into the TF command.

This might pose a challenge if running locally?

Reply:
added the arm_use_msi
I think for local use we can have arm_use_msi equal to false and supply the client id + secret